recipes/mysql-concpp/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+                                    strict=False)
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+    def package_info(self):
+        # Set the VS version
+        if self.settings.os == "Windows":
+            msvc_toolset_version = self.settings.get_safe("compiler.toolset")
+            msvc_version = self.settings.get_safe("compiler.version")
+
+            if msvc_toolset_version:
+                vs_version = msvc_toolset_version[1:3]
+            elif msvc_version:
+                vs_version = msvc_version[:2]
+                if vs_version == "18":
+                    vs_version = "12"
+                elif vs_version == "19":
+                    vs_version = "14"
+            else:
+                self.output.warn("MSVC_TOOLSET_VERSION or MSVC_VERSION not defined")
+                vs_version = "unknown"
+
+            self.output.info(f"VS Version: {vs_version}")
+            self.vs_version = vs_version
+            self.vs = f"vs{vs_version}"
+        else:
+            self.vs = ""
+
+        template_dirs = ["lib64", "lib64/debug", "lib", "lib/debug"]
+        self.cpp_info.libdirs = template_dirs if not self.vs else [f"{lib}/{self.vs}" for lib in template_dirs]
+        self.cpp_info.bindirs = template_dirs
+
+        if is_apple_os(self) or self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.extend(["resolv"])
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs.extend(["m", "crypto", "ssl"])
+
+        target = "concpp-xdevapi"
+        target_alias = "concpp"
+
+        if self.options.shared:
+            target += "-static"
+            target_alias += "-static"
+
+        if self.settings.build_type == "Debug":
+            target += "-debug"
+            target_alias += "-debug"
+
+        # self.cpp_info.set_property("cmake_target_name", f"mysql::{target}")
+        self.cpp_info.set_property("cmake_target_name", "mysql::concpp")
+        self.cpp_info.set_property("cmake_target_aliases", [f"mysql::{target_alias}"] )
+
+        lib = "mysqlcppconnx" if self.options.shared else "mysqlcppconnx-static"
+        if is_msvc(self) and not self.options.shared and is_msvc_static_runtime(self):
+            lib += "-mt"
+        self.cpp_info.libs = [lib]
+
+        if not self.options.shared:
+            self.cpp_info.defines = ["MYSQL_STATIC"]
+            self.cpp_info.defines = ["STATIC_CONCPP"]
+
+        self.cpp_info.set_property("cmake_find_package", "mysql-concpp")
+        self.cpp_info.set_property("cmake_find_package_multi", "mysql-concpp")
+        self.cpp_info.set_property("cmake_config_file", "mysql-concpp-config.cmake")
```

Comment:
```suggestion
```
These won't work and should be `self.cpp_info.names["cmake_find_package"] = "..."` instead. Would need extra handling of mismatched package and namespace names. Not worth the effort. I'd simply drop any legacy generators support.